sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/Column.java

```diff
@@ -82,6 +98,15 @@ static Column create(
   @Nullable
   ColumnDefaultValue defaultValue();
 
+  /**
+   * Returns the generation expression of this table column. Null means no generation expression.
+   * <p>
+   * The generation expression is stored as spark SQL dialect. It is up to the data source to verify
+   * expression compatibility and reject writes as necessary.
+   */
+  @Nullable
+  String generationExpression();
```

Comment:
I've added it as just a string here as it seems we don't need any additional information